cpp/src/arrow/compute/api_vector.cc

```diff
@@ -383,6 +401,14 @@ Result<Datum> CumulativeSum(const Datum& values, const CumulativeSumOptions& opt
   return CallFunction(func_name, {Datum(values)}, &options, ctx);
 }
 
+Result<Datum> CumulativeProduct(const Datum& values,
+                                const CumulativeProductOptions& options,
+                                ExecContext* ctx) {
+  auto func_name =
+      (options.check_overflow) ? "cumulative_product_checked" : "cumulative_product";
+  return CallFunction(func_name, {Datum(values)}, &options, ctx);
```

Comment:
May be

```suggestion
  return CallFunction(
      options.check_overflow ? "cumulative_sum_checked" : "cumulative_sum",
      {Datum(values)}, &options, ctx);
```

Reply:
if not we can just remove the parenthesis, 

```suggestion
  auto func_name =
      options.check_overflow ? "cumulative_product_checked" : "cumulative_product";
  return CallFunction(func_name, {Datum(values)}, &options, ctx);
```

Reply:
It's not necessary but not a big deal, either (surprised the formatter doesn't catch it though)